agreement/gossip/network.go

```diff
@@ -38,6 +39,10 @@ var (
 var messagesHandled = metrics.MakeCounter(metrics.AgreementMessagesHandled)
 var messagesDropped = metrics.MakeCounter(metrics.AgreementMessagesDropped)
 
+// Trace is the client for sending trace records for network propagation statistics.
+// It is public to allow configuration to be pushed in from node/node.go
+var Trace messagetracer.MessageTracer
```

Comment:
could you move this to the `networkImpl` and make it non-public ?
The node is also the one that calls the WrapNetwork, so it would be trivial to pass it in right after creation.